### ISSUE TITLE
Keras prints wrong values for metric.

### ISSUE THREAD

**Issue by somedadaism** [keras-team/keras#11882]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by somedadaism** [comment]
I have written a class for training a vae gan.
This is one of two loops, training the Generator network.
The train and test generator a custom made and return `(x_batch, x_batch)` after stetting   
`modus = encoding_modus`.  
```
class GeneratorTrainig:
    def __init__(self, discriminating_net, generating_net, train_generator, test_generator):
        self.discriminating_net = discriminating_net
        self.generating_net = generating_net
        self.train_generator = train_generator
        self.test_generator = test_generator
        def discrimination_loss(X_true, X_pred):
            d_true = self.discriminating_net(X_true)
            d_pred = self.discriminating_net(X_pred)
            return tf.log(d_true[0]) + tf.log(1-d_pred[1])
        self.discrimination_loss = discrimination_loss
        def acc_metric(X_true, X_pred):
            #d_true = self.discriminating_net(X_true)
            d_pred = self.discriminating_net(X_pred)
            return d_pred[0]
        self.acc_metric = acc_metric

    def __enter__(self):
        self.discriminating_net.train_able = False
        self.test_generator.modus = encoding_modus
        self.train_generator.modus = encoding_modus
        return self

    def __exit__(self, type, value, traceback):
        self.discriminating_net.train_able = True
        self.test_generator.modus = discriminating_modus_from(self.generating_net)
        self.train_generator.modus = discriminating_modus_from(self.generating_net)

    def fit(self, epochs):
        self.generating_net.compile(optimizer=SGD(clipnorm=10), loss=self.discrimination_loss, metrics =[self.acc_metric])
        self.generating_net.fit_generator(self.train_generator, epochs=epochs,
                                          callbacks=[WeightCheckpoint('modeling/model_weights/generator_weights.h5',
                                                     period=100)])
```
The class is used like this:
```
with GeneratorTrainig(discriminating_net, generating_net, train_generator, test_generator) as loop:
    loop.fit(epochs=10)
```

Now the issue is that I have highly confusing behavior with the training:
```
Epoch 2/10

1/4 [======>.......................] - ETA: 26s - loss: -1.4656 - acc_metric: 0.5000
2/4 [==============>...............] - ETA: 17s - loss: -1.4682 - acc_metric: 0.5000
3/4 [=====================>........] - ETA: 6s - loss: -1.4687 - acc_metric: 0.5000 
4/4 [==============================] - 28s 7s/step - loss: -1.4681 - acc_metric: 0.5000
```
```
1/4 [======>.......................] - ETA: 33s - loss: -1.4711 - acc_metric: 0.5000
2/4 [==============>...............] - ETA: 12s - loss: -1.4702 - acc_metric: 0.5000
3/4 [=====================>........] - ETA: 7s - loss: -1.4713 - acc_metric: 0.5000 
4/4 [==============================] - 34s 9s/step - loss: -1.4711 - acc_metric: 0.5000
```
I can not understand how this behavior is even mathematically possible.
